spirv_msl.cpp

```diff
@@ -1466,9 +1480,32 @@ void CompilerMSL::mark_as_packable(SPIRType &type)
 }
 
 // If a shader input exists at the location, it is marked as being used by this shader
-void CompilerMSL::mark_location_as_used_by_shader(uint32_t location, StorageClass storage)
+void CompilerMSL::mark_location_as_used_by_shader(uint32_t location, const SPIRType &type, StorageClass storage)
 {
-	if (storage == StorageClassInput)
+	if (storage != StorageClassInput)
+		return;
+	if (is_array(type))
+	{
+		for (uint32_t i = 0; i < type.array.size(); i++)
```

Comment:
Seems a bit cleaner to multiply together all dimensions and then loop over number of dimensions separately.